primo/opt_model/opt_model.py

```diff
@@ -419,6 +427,38 @@ def project_max_spend_rule(model, campaign_id):
                 doc="Define limit on max project spend",
             )
             LOGGER.info("Added max project spend constraint")
+
+        def budget_constraint_slack(model):
+            """
+            Implement a slack variable to maximize the utilization of the total budget.
+            The constraint intends to avoid the scenario where there is no
+            wells being selected.
+
+            Parameters
+            ----------
+            model : pyo.ConcreteModel
+                The optimization model.
+
+            Returns
+            -------
+            pyo.Constraint
+            """
+            return (
+                model.p_B
+                - sum(
+                    model.v_q[cluster, n_well] * model.p_c[n_well]
+                    for (cluster, n_well) in model.v_q.index_set()
+                )
+                <= model.B_slack
+            )
+
+        model.con_budget_slack = pyo.Constraint(
+            rule=budget_constraint_slack,
+            doc="A slack variable for unused budget",
+        )
+
+        LOGGER.info("Added a budget constraint to force as much as budget being used")
```

Comment:
"Added a budget..." -> "Added constraint to calculate the amount of budget that is not utilized"